stdlib/public/core/String.swift

```diff
@@ -584,24 +584,27 @@ extension String {
 #else
     switch (_core.isASCII, rhs._core.isASCII) {
     case (true, false):
-      let lhsPtr = UnsafePointer<Int8>(_core.startASCII)
-      let rhsPtr = UnsafePointer<UTF16.CodeUnit>(rhs._core.startUTF16)
+      let lhsPtr = unsafeBitCast(_core.startASCII,
+        to: UnsafePointer<CChar>.self)
+      let rhsPtr = UnsafePointer(rhs._core.startUTF16)
 
       return Int(_swift_stdlib_unicode_compare_utf8_utf16(
         lhsPtr, Int32(_core.count), rhsPtr, Int32(rhs._core.count)))
     case (false, true):
       // Just invert it and recurse for this case.
       return -rhs._compareDeterministicUnicodeCollation(self)
     case (false, false):
-      let lhsPtr = UnsafePointer<UTF16.CodeUnit>(_core.startUTF16)
-      let rhsPtr = UnsafePointer<UTF16.CodeUnit>(rhs._core.startUTF16)
+      let lhsPtr = UnsafePointer(_core.startUTF16)
+      let rhsPtr = UnsafePointer(rhs._core.startUTF16)
```

Comment:
startUTF16 is mutable. Why would implicit conversions be activated? AFAIK that's only for argument passing.